### PR TITLE
Adjust to signature help changes in scala3 compiler

### PR DESCRIPTION
**Pull request by dos65**
Fixes tests failures in latest nightly
@tgodzik I rebased your branch

### CODE REVIEW

**Comment by dos65** [comment]
The new method `Signatures.signatureHelp` works only from 2022-06-10 nightly while `test download dependencies` tries to compile 5 last nighty releases(which includes 2022-06-09 that doesn't have this method yet)

I think it's easier to merge it as it is. This issue with test will disappear after one more nightly version will be released.

File: mtags/src/main/scala-3.2/scala/meta/internal/pc/MetalsSignatures.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+object MetalsSignatures:
+
+  def signatures(
+      search: SymbolSearch,
+      path: List[tpd.Tree],
+      pos: SourcePosition
+  )(using ctx: Context): (Int, Int, List[(Signature, Denotation)]) =
+    val (paramN, callableN, alternatives) =
+      Signatures.signatureHelp(path, pos.span)
```
Comment (tanishiking) [inline mtags/src/main/scala-3.2/scala/meta/internal/pc/MetalsSignatures.scala:21]: [note]
`callInfo` is deprecated in favor of `signatureHelp`
https://github.com/lampepfl/dotty/pull/15278

**Review by tanishiking — APPROVED** [review]
The change itself is LGTM:tada: (except cross/test looks failing).
Also, can you add comments in code or commit message that why do we have this change pointing to the issue https://github.com/lampepfl/dotty/issues/15244 ?

**Comment by tanishiking** [comment]
:+1: to commit message

**Comment by dos65** [comment]
hmm. latest nightly cross test fails again :disappointed:

**Pull request merged**